include/boost/url/pct_encoded_view.hpp

```diff
@@ -0,0 +1,696 @@
+//
+// Copyright (c) 2022 Alan Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CPPAlliance/url
+//
+
+#ifndef BOOST_URL_PCT_ENCODED_VIEW_HPP
+#define BOOST_URL_PCT_ENCODED_VIEW_HPP
+
+#include <boost/url/detail/config.hpp>
+#include <boost/url/string_view.hpp>
+#include <boost/url/pct_encoding.hpp>
+#include <type_traits>
+#include <iterator>
+#include <iosfwd>
+
+namespace boost {
+namespace urls {
+
+/** A view of the percent decoded characters in a string
+
+    This class is a lightweight wrapper that presents
```

Comment:
Instead of all this lightweight wrapper business you could just say

Objects of type pct_encoded_view reference URL or other component strings that are percent-encoded. That is, that characters which are special (not in the allowed character set) are represented with escapes that start with a percent followed by a two-digit hexadecimal number of the corresponding character code (which may be part of a UTF-8 code point depending on the context). The primary operations are:

* Comparison to other percent-encoded or plain strings
* Appending the decoded characters to an existing container
* Assigning the decoded characters to an existing container
* Iterating over the represented decoded character range
* Accessing the underlying percent-encoded character buffer
...